scan_engine.cc

```diff
@@ -2031,6 +2032,20 @@ static bool ultrascan_host_pspec_update(UltraScanInfo *USI, HostScanStats *hss,
   if (hss->target->flags != HOST_UP) {
     assert(newstate == HOST_UP || newstate == HOST_DOWN);
     hss->target->flags = newstate;
+  /* Log if the host is up and this is non pure host discovery scan. */
+    if (oldstate != newstate && newstate == HOST_UP && o.noportscan == false) {
```

Comment:
so `o.noportscan` is how well tell it is a host discovery scan then?

Presumably just this file change we require + the dtd and the other changes can be dropped as unrelated?